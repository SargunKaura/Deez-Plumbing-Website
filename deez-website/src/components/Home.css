body {
        margin: 0;
        font-family: Arial, sans-serif;
      }
      
      .App {
        text-align: center;
        color: #333;
      }
      
      .header {
        display: flex;
        justify-content: space-between;
        padding: 1rem;
        background-color: #e0f7fa;
      }
      
      .header h1 {
        margin: 0;
      }
      
      .header nav a {
        margin: 0 1rem;
        text-decoration: none;
        color: #00796b;
      }
      
      .hero {
        padding: 2rem;
        background-color: #a7ffeb;
      }
      
      .hero h2 {
        margin: 0.5rem 0;
      }
      
      .appointment-button {
        margin-top: 1rem;
        padding: 0.5rem 1rem;
        background-color: #00796b;
        color: white;
        border: none;
        cursor: pointer;
      }
      
      .hours-operation, .mission, .reviews, .contact {
        padding: 2rem;
        border-bottom: 1px solid #e0e0e0;
      }
      
      .mission-button {
        margin-top: 1rem;
        padding: 0.5rem 1rem;
        background-color: #00796b;
        color: white;
        border: none;
        cursor: pointer;
      }
      
      .reviews .review-images {
        display: flex;
        justify-content: center;
        gap: 2rem;
      }
      
      .reviews .testimonials {
        display: flex;
        justify-content: center;
        gap: 2rem;
        margin-top: 1rem;
      }
      
      .reviews .testimonial {
        border: 1px solid #e0e0e0;
        padding: 1rem;
        background-color: #ffffff;
        box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
      }
      
      .review-button {
        margin-top: 1rem;
        padding: 0.5rem 1rem;
        background-color: #00796b;
        color: white;
        border: none;
        cursor: pointer;
      }
      
      .contact p {
        margin: 0.5rem 0;
      }
      
      .footer {
        padding: 1rem;
        background-color: #e0f7fa;
      }
      
      .footer nav a {
        margin: 0 1rem;
        text-decoration: none;
        color: #00796b;
      }
      