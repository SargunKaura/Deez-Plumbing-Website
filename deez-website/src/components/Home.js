import './Home.css';
import { Link } from 'react-router-dom'; // Import Link from react-router-dom
import React from 'react';

function Home() {
  return (
    <div className="App">

      <header>
        <Link to="/">
          <h1>DPL</h1>
        </Link>
        <Link to="/">
          <button>Contact Us</button>
        </Link>
        <Link to="/OurMission">
          <button>Our Mission</button>
        </Link>
        <Link to="/BookAppointment">
          <button>Book Appointment</button>
        </Link>
      </header>

      <h2>Deez Plumbing LTD</h2>

      <h4>Local Plumbers for Local People.</h4>

      <p>
        Hours of operation<br />
        Weekdays: 4:00 PM - 11:00 PM<br />
        Weekends: 5:30 AM - 11:00 PM<br />
        Holidays: CLOSED
      </p>

      <Link to="/BookAppointment">
        <button>Book Appointment</button>
      </Link>

      <h3>Reviews and Testimonials</h3>

      <a href="https://www.google.com/search?q=Deez+Plumbing+LTD+reviews" target="_blank" rel="noopener noreferrer">
        <button>Read More</button>
      </a>

      <h3>Our Mission</h3>
      <p>Two apprentice plumbers pursuing their dreams, founding their own company with a vision for excellence in plumbing services and customer satisfaction.</p>

      <Link to="/OurMission">
        <button>Meet the Crew</button>
      </Link>

      <h3>Contact Us</h3>

      <p>
        Email: Deezplumbers@gmail.com<br></br>
        Phone: TBD
      </p>
      
    </div>
  );
}

export default Home;
