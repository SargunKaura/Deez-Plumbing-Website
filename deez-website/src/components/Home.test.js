import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from './Home';

test('renders Home component and matches snapshot', () => {
    const component = renderer.create(
        <Router>
            <Home />
        </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
