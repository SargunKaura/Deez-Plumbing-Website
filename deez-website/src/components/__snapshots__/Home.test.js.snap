// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component and matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <div
      className="DPL-header"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="DPL-header-button"
        >
          DPL
        </button>
      </a>
    </div>
    <div
      className="buttons-header-DIV"
    >
      <a
        href="/BookAppointment"
        onClick={[Function]}
      >
        <button
          className="header-button"
        >
          Book an Appointment
        </button>
      </a>
      <button
        className="header-button"
        onClick={[Function]}
      >
         Contact Us
      </button>
      <a
        href="/OurMission"
        onClick={[Function]}
      >
        <button
          className="header-button"
        >
          Our Mission
        </button>
      </a>
    </div>
  </header>
  <main>
    <section
      className="title"
    >
      <h2>
        Deez Plumbing LTD
      </h2>
      <p>
        Local Plumbers for Local People.
      </p>
      <a
        href="/BookAppointment"
        onClick={[Function]}
      >
        <button
          className="main-button"
          id="appointment-button"
        >
          Book an Appointment
        </button>
      </a>
    </section>
    <section
      className="hours-operation"
    >
      <div
        className="hours-content"
      >
        <h3>
          Hours of Operation
        </h3>
        <p>
          Weekdays: 4:00 PM - 11:00 PM
        </p>
        <p>
          Weekends: 5:30 AM - 11:00 PM
        </p>
        <p>
          Holidays: CLOSED
        </p>
      </div>
      <div
        className="carousel-div"
      >
        <button
          className="carousel-button"
          onClick={[Function]}
        >
          ← 
        </button>
        <div
          className="carousel"
        >
          <div
            className="carousel-slide"
            style={
              Object {
                "transform": "translateX(-0%)",
                "transition": "transform 0.3s ease-in-out",
              }
            }
          >
            <img
              alt="Slide 0"
              src="/images/hours-image1.jpg"
            />
            <img
              alt="Slide 1"
              src="/images/hours-image2.jpg"
            />
            <img
              alt="Slide 2"
              src="/images/hours-image3.jpg"
            />
            <img
              alt="Slide 3"
              src="/images/hours-image4.jpg"
            />
          </div>
        </div>
        <button
          className="carousel-button"
          onClick={[Function]}
        >
          → 
        </button>
      </div>
    </section>
    <section
      className="mission"
    >
      <div
        className="mission-text"
      >
        <h3>
          Our Mission
        </h3>
        <p>
          Two apprentice plumbers pursuing their dreams, founding their own company with a vision for excellence in plumbing services and customer satisfaction.
        </p>
      </div>
      <div
        className="mission-elements"
      >
        <a
          href="/OurMission"
          onClick={[Function]}
        >
          <button
            className="main-button"
            id="mission-button"
          >
            Meet the Crew
          </button>
        </a>
      </div>
    </section>
    <section
      className="reviews"
    >
      <div
        className="reviews-text"
      >
        <h3>
          Reviews and Testimonials
        </h3>
        <p>
          Our work and reviews.
        </p>
      </div>
      <div
        className="carousel-div"
      >
        <button
          className="carousel-button"
          onClick={[Function]}
        >
          ← 
        </button>
        <div
          className="carousel"
        >
          <div
            className="carousel-slide"
            style={
              Object {
                "transform": "translateX(-0%)",
                "transition": "transform 0.3s ease-in-out",
              }
            }
          >
            <img
              alt="Slide 0"
              src="/images/default-image.jpg"
            />
            <img
              alt="Slide 1"
              src="/images/default-image.jpg"
            />
            <img
              alt="Slide 2"
              src="/images/default-image.jpg"
            />
            <img
              alt="Slide 3"
              src="/images/default-image.jpg"
            />
          </div>
        </div>
        <button
          className="carousel-button"
          onClick={[Function]}
        >
          → 
        </button>
      </div>
      <div
        className="testimonials"
      >
        <div
          className="testimonial"
        >
          <p>
            "A heroic piece of praise!"
          </p>
          <p>
            - Name
          </p>
        </div>
        <div
          className="testimonial"
        >
          <p>
            "A fantastic bit of feedback!"
          </p>
          <p>
            - Name
          </p>
        </div>
        <div
          className="testimonial"
        >
          <p>
            "A genuinely glowing review!"
          </p>
          <p>
            - Name
          </p>
        </div>
      </div>
      <div
        className="review-button-Div"
      >
        <button
          className="main-button"
          id="review-button"
        >
          Read more Reviews
        </button>
      </div>
    </section>
    <section
      className="contact"
      id="contact-section"
    >
      <h3>
        Contact Us
      </h3>
      <p>
        Email: Deezplumbers@gmail.com
      </p>
      <p>
        Business phone: TBD
      </p>
    </section>
  </main>
  <footer
    className="footer"
  >
    <div
      className="DPL-footer"
    >
      <p>
        Deez Plumbing LTD
      </p>
    </div>
    <nav>
      <div
        className="footer-links"
      >
        <a
          href="#our-mission"
        >
          Our Mission
        </a>
        <a
          href="#book-appointment"
        >
          Book an Appointment
        </a>
      </div>
    </nav>
  </footer>
</div>
`;
